pkgs/applications/networking/browsers/qutebrowser/default.nix

```
@@ -86,7 +83,7 @@ python3.pkgs.buildPythonApplication {
     runHook preInstall
 
     make -f misc/Makefile \
-      PYTHON=${buildPackages.python3}/bin/python3 \
+      PYTHON=${python3.pythonForBuild.interpreter} \
```

`pkgsForBuild` is be wrong here, because we really want python from `pkgsOnBuildForHost`.

Your suggested change works only because, confusingly,

```
python.pythonForBuild != pkgsForBuild.python
```

... there's a mistake *in the opposite direction* in pythons' `passthrufuns.txt` (with a `TODO` saying to correct it!) that undoes the error:

https://github.com/NixOS/nixpkgs/blob/eecc84500f0dc82f3b8b852a55dff494578766a2/pkgs/development/interpreters/python/passthrufun.nix#L92-L94

Let's not rely on this please.

So I suggest:

```suggestion
      PYTHON=${python3.pythonOnBuildForHost.interpreter} \
```

which may require cherry-picking 563a66dc314ec75fe2c936f80d83005899b0ed9c from https://github.com/NixOS/nixpkgs/pull/265570.  Let's not perpetuate the mistake in `passthrufuns.txt`...

---

> ... there's a mistake in the opposite direction in pythons' passthrufuns.txt (with a TODO saying to correct it!) that undoes the error:

It's really just a naming issue, not a functional issue

---

Yeah but there's a reason why people say our cross-compilation infrastructure is too confusing...  inconsistent naming is a big part of that.